### PR TITLE
chore(updatecli): add a manifest to update azure service principal password for the trusted controller

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/4107

track the expiration date for the controller azure service principal on trusted

### CODE REVIEW

**Comment by smerle33** [comment]
```
-  controller_service_principal_end_date = "2024-06-09T19:40:35Z"
+  controller_service_principal_end_date = "2024-08-27T00:00:00Z"
...
⚠ Generate new end date for trusted.ci.jenkins.io controller service principal:
        Source:
                ✔ [currentExpiry] Get current `expiry` date
                ✔ [nextExpiry] Prepare next `expiry` date within 3 months
        Condition:
                ✔ [checkIfExpirySoonExpired] 
        Target:
                ⚠ [updateNextExpiry] Update Terraform file `trusted.ci.jenkins.io.tf` with new expiration date for controller_service_principal_end_date


Run Summary
===========
Pipeline(s) run:
  * Changed:    1
  * Failed:     0
  * Skipped:    0
  * Succeeded:  0
  * Total:      1
 ```

File: updatecli/updatecli.d/trusted.ci.sp.endate.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: "Generate new end date for trusted.ci.jenkins.io controller service principal"
```
Comment (dduportal) [inline updatecli/updatecli.d/trusted.ci.sp.endate.yaml:1]:
```suggestion
name: "Generate new end date for the trusted.ci.jenkins.io controller Azure AD Application password"
```

Comment (dduportal) [reply updatecli/updatecli.d/trusted.ci.sp.endate.yaml:1]: It's an Azure AD password, not a Service Principal password as per https://github.com/jenkins-infra/shared-tools/blob/adae7d43808f77c2a94bac3846d6219da93b3261/terraform/modules/azure-jenkinsinfra-controller/main.tf#L303-L307

File: updatecli/updatecli.d/trusted.ci.sp.endate.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      matchpattern: (controller_service_principal_end_date.*=.*\")(.*)(\".*)
+      replacepattern: ${1}{{ source "nextExpiry" }}"
+    scmid: default
+
+actions:
+  default:
+    kind: github/pullrequest
+    scmid: default
+    spec:
+      title: Generate a new Azure Service Principal password with expiration date {{ source "nextExpiry" }} on `trusted.ci.jenkins.io`
```
Comment (dduportal) [inline updatecli/updatecli.d/trusted.ci.sp.endate.yaml:55]:
```suggestion
      title: Generate a new Azure AD Application password with expiration date {{ source "nextExpiry" }} on `trusted.ci.jenkins.io`
```

It's an Azure AD password, not a Service Principal password as per https://github.com/jenkins-infra/shared-tools/blob/adae7d43808f77c2a94bac3846d6219da93b3261/terraform/modules/azure-jenkinsinfra-controller/main.tf#L303-L307

File: updatecli/updatecli.d/trusted.ci.sp.endate.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      description: |
+        Generate a new password for the Azure Service Principal `trusted.ci.jenkins.io`.
+        once this PR is merged and deployed with success by Terraform on infra.ci.jenkins.io , and the new Azure AD password's values has been retrieved to a secured (encrypted) location.
+        update manually trusted.ci.jenkins.io's credential `azure-jenkins-sponsorship-credentials`` through the UI
+        Verified both the Azure Credential and Azure VM clouds by clicking on the "Verify <...>" buttons with success
```
Comment (dduportal) [inline updatecli/updatecli.d/trusted.ci.sp.endate.yaml:60]:
```suggestion
        This PR generates a new Azure AD application password with a new end date for the `trusted.ci.jenkins.io` controller (to allow spawning Azure VM agents).
        Once this PR is merged and deployed with success by Terraform (on infra.ci.jenkins.io), you can retrieve the new password's value in Terraform state (do NOT save it anywhere!) and update (manually) the trusted.ci.jenkins.io credential named `azure-jenkins-sponsorship-credentials` through the Jenkins UI.
        Finally, verify both the Azure Credential and Azure VM clouds by clicking on the "Verify <...>" buttons with success.
```

**Review by dduportal — APPROVED** [review]
Good work! A few wording elements: not blocking strictly speaking but I would prefer to get these wording changes for clearer PRs.

**Pull request merged**